Block scaffolding: switch to namespaces for block registration

jeherve opened the pull request:
#### Changes proposed in this Pull Request:

* This should make things clearer for future blocks, with less repetitions and more opportunities when folks extend that basic php file where the block is registered. It also provides the basic tools for block gating as it's done for some of our blocks already.

#### Testing instructions:

* Run `yarn docker:wp jetpack scaffold block "Cool block"`
* Open the editor
* Try adding a "Cool block" to the editor.
* Check that the block loads correctly

#### Proposed changelog entry for your changes:

* N/A

jeherve committed:
Block scaffolding: switch to namespaces for block registration

This should make things clearer for future blocks.

scruffian commented on wp-cli-templates/block-register-php.mustache:
```diff
@@ -7,10 +7,23 @@
-	'jetpack/{{ slug }}',
-	array( 'render_callback' => 'jetpack_{{ underscoredSlug }}_block_load_assets' )
-);
+namespace Jetpack\{{ title }}_Block;
```
I wonder if this should be {{ slug }} too?

jeherve replied:
Yes you're right, it needs to account for block names that may include spaces. c58458d302f701c147e774111bbaac51936bc516 should take care of that.

jeherve committed:
Remove spaces from block names for namespace declaration

See https://github.com/Automattic/jetpack/pull/14728#discussion_r381223058

In those situations, we need a slug with each word capitalized. Since $title already comes with each word capitalized, we can use it as a starter.

scruffian reviewed (approved):
LGTM

kbrown9 reviewed (approved):
This looks good! It tests well. The expected block files were created, and the block loads in the editor.

Note for anyone else that wants to test this and isn't very familiar with testing blocks - you need to run `yarn run build-extensions` to build the block before you can add the block in the editor.
